libraries/SITL/SIM_Precland.cpp

```diff
@@ -112,24 +119,30 @@ void SIM_Precland::update(const Location &loc, const Vector3d &position)
         return;
     }
 
-    const float distance_z = -position.z;
-    const float origin_height_m = _origin_height * 0.01f;
-    if (distance_z > _alt_limit + origin_height_m) {
-        _healthy = false;
-        return;
-    }
-
     const Location origin_center(static_cast<int32_t>(_origin_lat * 1.0e7f),
             static_cast<int32_t>(_origin_lon * 1.0e7f),
             static_cast<int32_t>(_origin_height),
             Location::AltFrame::ABOVE_HOME);
-    Vector2f center;
-    if (!origin_center.get_vector_xy_from_origin_NE(center)) {
+    Vector3f center;
+    if (!origin_center.get_vector_from_origin_NEU(center)) {
+        _healthy = false;
+        return;
+    }
+    center = center * 0.01f;        // cm to m
+    center.z = -center.z;           // neu to ned
+    Vector3d axis{1, 0, 0};
+    axis.rotate((Rotation)_orient.get());   // unit vector in direction of axis of cone or cylinder
+    Vector3d position_wrt_origin(position.x - center.x, position.y - center.y, position.z - center.z);  // position of vehicle with respect to preland device origin
+    const float longitudinal_dist = position_wrt_origin.projected(axis).length();   // longitudinal distance of vehicle from precland device
+    const float lateral_distance = safe_sqrt(MAX(0, position_wrt_origin.length_squared() - longitudinal_dist*longitudinal_dist));   // lateral distance of vehicle from precland device
+    _over_precland_base = lateral_distance <= 2.0f;
+
+    // return false if vehicle if vehicle is  longitudinally too far away from precland device
```

Comment:
Might want to comment further up in here what you're referring to be lateral and longitudinal distance.

Reply:
Comments added.  Thanks.